plugins/kn-image/cmd/build.go

```diff
@@ -0,0 +1,114 @@
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc" // from https://github.com/kubernetes/client-go/issues/345
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+)
+
+const (
+	// How often to retry in case of an optimistic lock error when replacing a service (--force)
+	MaxUpdateRetries = 3
+	// Timeout to wait service creation
+	MaxTimeout = 300
+)
+
+// buildCmd represents the build command
+var buildCmd = &cobra.Command{
+	Use:   "build",
+	Short: "Build the container image from source code",
+	Example: `
+  # Build from git repository into an image
+  # ( related: https://github.com/knative/client-contrib/blob/master/plugins/kn-image/doc/deploy-git-resource.md )
+  kn-image build example-image --git-url https://github.com/bluebosh/knap-example -git-revision master --builder kaniko --saved-image us.icr.io/test/example-image --serviceaccount default`,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("")
+		if len(args) < 1 {
+			cmd.Help()
+			os.Exit(0)
+		}
+		name := args[0]
+
+		fmt.Println("[INFO] Build from git repository into an image")
+		serviceAccount := cmd.Flag("serviceaccount").Value.String()
+		namespace := cmd.Flag("namespace").Value.String()
+
+		// Config kubeconfig
+		kubeconfig = cmd.Flag("kubeconfig").Value.String()
+		if kubeconfig == "" {
+			kubeconfig = os.Getenv("KUBECONFIG")
+		}
+		cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			fmt.Println("[ERROR] Parsing kubeconfig error:", err)
+			os.Exit(1)
+		}
+		client, err := tektoncdclientset.NewForConfig(cfg)
+		if err != nil {
+			fmt.Println("[ERROR] Building kubeconfig error:", err)
+			os.Exit(1)
+		}
+		tektonClient := tekton.NewTektonClient(client.TektonV1alpha1(), namespace)
+
+		builder := cmd.Flag("builder").Value.String()
+		if builder == "" {
+			fmt.Println("[ERROR] Builder cannot be empty, please use --builder to set")
+			os.Exit(1)
+		}
+
+		gitUrl := cmd.Flag("git-url").Value.String()
+		if gitUrl == "" {
+			fmt.Println("[ERROR] Git url cannot be empty, please use --git-url to set")
+			os.Exit(1)
+		}
+		gitRevision := cmd.Flag("git-revision").Value.String()
+		if gitRevision == "" {
+			fmt.Println("[ERROR] Git revision cannot be empty, please use --git-revision to set")
+			os.Exit(1)
+		}
+		image := cmd.Flag("saved-image").Value.String()
+		if image == "" {
+			fmt.Println("[ERROR] Saved-image cannot be empty, please use --saved-image to set")
+			os.Exit(1)
+		}
+
+		if len(gitUrl) > 0 {
+			err = tektonClient.BuildFromGit(name, builder, gitUrl, gitRevision, image, serviceAccount, namespace)
+			if err != nil {
+				fmt.Println("[ERROR] Building image error:", err)
+				os.Exit(1)
+			}
+			fmt.Println("[INFO] Generate image", image, "from git repo")
+		}
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(buildCmd)
+
+	buildCmd.Flags().StringP("builder", "b", "", "builder of source-to-image task")
+	buildCmd.Flags().StringP( "git-url", "u","", "[Git] url of git repo")
```

Comment:
Why “[Git]” instead of just “Git” or “git”?

Reply:
I add `[Git]` to tell user this option is for git resource only.
In future, we may support different options like [Local] or [Function]